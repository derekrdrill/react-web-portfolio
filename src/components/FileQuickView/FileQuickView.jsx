import React from 'react';
import PropTypes from 'prop-types';
import FileViewer from 'react-file-viewer';
import styled from 'styled-components';
import { Button, Modal } from '@mui/material';
import { Container, Row, Col } from 'react-bootstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faDownload } from '@fortawesome/fontawesome-free-solid';
import { faGoogleDrive } from '@fortawesome/free-brands-svg-icons';

export const FileQuickView = ({ fileType, filePath, onlineLink, open, onClose }) => (
  <Modal open={open} onClose={onClose}>
    <FileQuickViewContainer fluid>
      <FileQuickViewToolbar>
        <Col xs={1} className='mt-1'>
          <Button color='error' variant='text' onClick={onClose}>
            EXIT
          </Button>
        </Col>
        <Col xs={{ span: 1, offset: 10 }}>
          <UtilityButtons href={filePath} download>
            <FontAwesomeIcon icon={faDownload} />
          </UtilityButtons>
          <UtilityButtons href={onlineLink} target='_blank'>
            <FontAwesomeIcon icon={faGoogleDrive} />
          </UtilityButtons>
        </Col>
      </FileQuickViewToolbar>
      <Row>
        <FileViewer fileType={fileType} filePath={filePath} />
      </Row>
    </FileQuickViewContainer>
  </Modal>
);

FileQuickView.propTypes = {
  filePath: PropTypes.string,
  fileType: PropTypes.string,
  onClose: PropTypes.func,
  onlineLink: PropTypes.string,
  open: PropTypes.bool,
};

const FileQuickViewContainer = styled(Container)({
  height: '85%',
  width: '55%',
  overflow: 'auto',
  border: '1px solid black',
  borderRadius: 3,
  transform: 'translateY(5vh)',
  backdropFilter: 'blur(10px)',
  canvas: {
    width: '92%',
    marginLeft: '4%',
  },
});

const FileQuickViewToolbar = styled(Row)({
  backgroundColor: 'gainsboro',
  position: 'sticky',
  top: 0,
  zIndex: 1,
});

const UtilityButtons = styled.a({
  padding: 8,
});
