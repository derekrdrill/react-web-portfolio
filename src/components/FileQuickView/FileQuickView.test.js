/*** @jest-environment jsdom */

import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { FileQuickView } from './FileQuickView';

const renderer = new ShallowRenderer();

describe('File Quick View tests', () => {
  it('renders correctly', () => {
    renderer.render(<FileQuickView open />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
