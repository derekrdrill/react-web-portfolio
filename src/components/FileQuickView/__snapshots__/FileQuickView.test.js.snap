// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File Quick View tests renders correctly 1`] = `
<ForwardRef(Modal)
  open={true}
>
  <Styled(Container)
    fluid={true}
  >
    <Styled(Row)>
      <Col
        className="mt-1"
        xs={1}
      >
        <ForwardRef(Button)
          color="error"
          variant="text"
        >
          EXIT
        </ForwardRef(Button)>
      </Col>
      <Col
        xs={
          Object {
            "offset": 10,
            "span": 1,
          }
        }
      >
        <styled.a
          download={true}
        >
          <FontAwesomeIcon
            beat={false}
            beatFade={false}
            border={false}
            bounce={false}
            className=""
            fade={false}
            fixedWidth={false}
            flip={false}
            icon={
              Object {
                "icon": Array [
                  512,
                  512,
                  Array [],
                  "f019",
                  "M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z",
                ],
                "iconName": "download",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            maskId={null}
            pull={null}
            pulse={false}
            rotation={null}
            shake={false}
            size={null}
            spin={false}
            spinPulse={false}
            spinReverse={false}
            swapOpacity={false}
            symbol={false}
            title=""
            titleId={null}
            transform={null}
          />
        </styled.a>
        <styled.a
          target="_blank"
        >
          <FontAwesomeIcon
            beat={false}
            beatFade={false}
            border={false}
            bounce={false}
            className=""
            fade={false}
            fixedWidth={false}
            flip={false}
            icon={
              Object {
                "icon": Array [
                  512,
                  512,
                  Array [],
                  "f3aa",
                  "M339 314.9L175.4 32h161.2l163.6 282.9H339zm-137.5 23.6L120.9 480h310.5L512 338.5H201.5zM154.1 67.4L0 338.5 80.6 480 237 208.8 154.1 67.4z",
                ],
                "iconName": "google-drive",
                "prefix": "fab",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            maskId={null}
            pull={null}
            pulse={false}
            rotation={null}
            shake={false}
            size={null}
            spin={false}
            spinPulse={false}
            spinReverse={false}
            swapOpacity={false}
            symbol={false}
            title=""
            titleId={null}
            transform={null}
          />
        </styled.a>
      </Col>
    </Styled(Row)>
    <Row>
      <t
        errorComponent={null}
        onError={[Function]}
        unsupportedComponent={null}
      />
    </Row>
  </Styled(Container)>
</ForwardRef(Modal)>
`;
